### PR TITLE
Mcafee mar improvements

### PR DESCRIPTION
**Pull request by anara123**
## Status
In Progress

## Description
Added projection_collector argument to search

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Additional changes
Describe additional changes done, for example adding a function to common server.

### CODE REVIEW

File: Tests/scripts/update_id_set.py [hunk 1]
```diff
@@ -283,16 +294,23 @@ def update_object_in_id_set(obj_id, obj_data, file_path, instances_set):
...
         integration_from_version = instance[instance_id].get('fromversion', '0.0.0')
+
         if obj_id == instance_id:
             if is_added_from_version or (not is_added_from_version and file_from_version == integration_from_version):
                 if is_added_to_version or (not is_added_to_version and file_to_version == integration_to_version):
                     instance[obj_id] = obj_data[obj_id]
+                    updated = True
                     break
 
+    if not updated:
```
[the remark this answers is not in the document]
Comment (anara123) [reply Tests/scripts/update_id_set.py]: I agree that it shouldn't happen, but for fix the problem I deleted Sentinel One from id_Set and update_id_set didnt created it back, that is why I added it. It is like edge case if we want to fix some indexing
